### PR TITLE
Frankenstein ligand tool for rdock workflow

### PR DESCRIPTION
**Pull request by simonbray**
Thanks a lot to @lorrainealisha75 for working on this.

### CODE REVIEW

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                yr = record[1]
+                zr = record[2]
+                if xr-tol < x and x < xr+tol and \
+                   yr-tol < y and y < yr+tol and \
+                   zr-tol < z and z < zr+tol:
+                    tv = 1
+                    break
+            if tv == 1:
+                continue
+            temp.append(x)
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:61]: temp.extend([x,y,z])

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    for molecule in all_coordinates:
+        for coordinates in molecule:
+            tv = 0
+            temp = []
+            x = coordinates[0]
+            y = coordinates[1]
+            z = coordinates[2]
+            for record in select:
+                xr = record[0]
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:51]: xr, yr, zr = record

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            temp.append(x)
+            temp.append(y)
+            temp.append(z)
+            select.append(temp)
+    return select
+
+
+def sdfout(centers, writer):
+    n = len(centers)
+    writer.write("MOL_1\n1234567890123456789 3D\nTITLE\n")
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:70]: shouldn't the name not be more meaningful?

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    writer.write("%3d  0  0  0  0  0  0  0  0  0999 V2000\n" % n)
+    for record in centers:
+        x = record[0]
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:73]: x,y,z = record

File: chemicaltoolbox/rdock/select_points_SDF.xml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<tool id="ctb_select_points" name="Create Frankenstein ligand" version="0.1.0">
+    <description>for rDock active site definition</description>
+    <requirements>
+        <requirement type="package" version="3.7">python</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '$__tool_directory__/select_points_SDF.py'
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.xml:7]: `python` should be enough

**Comment by simonbray** [comment]
> I guess this one does not need to be in the rdock folder?

No - seems the best place for it though IMO.

File: chemicaltoolbox/rdock/select_points_SDF.xml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<tool id="ctb_select_points" name="Create Frankenstein ligand" version="0.1.0">
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.xml:1]: @simonbray the ID should also be adopted, left? ctb_frankenstein_ligand?

Comment (bgruening) [reply chemicaltoolbox/rdock/select_points_SDF.xml:1]: select points seems to general for my taste

File: chemicaltoolbox/rdock/select_points_SDF.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+        
+.. class:: infomark
+
+**Output**
+
+       - SD file containing the 'Frankenstein ligand'.
+    ]]></help>
+       <citations>
+            <citation type="doi">10.1371/journal.pcbi.1003571</citation>
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.xml:55]: what is this I can not find it?

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                        help="Input file")
+    parser.add_argument('-o', '--output',
+                        help="Base name for output file (no extension).")
+    args = parser.parse_args()
+
+    mol_coordinates = []
+    all_coordinates = []
+    with open(args.input) as file:
+        for line in file:
+            if line == '$$$$\n':
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:79]: what happens with Windows line-ending ... should we use line.strip() == "$$$$"

File: chemicaltoolbox/rdock/select_points_SDF.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                temp = get_coordinates(mol_coordinates)
+                all_coordinates.append(temp)
+                mol_coordinates.clear()
+            else:
+                mol_coordinates.append(line)
+    centers = select_points(all_coordinates)
+    writer = open(args.output, 'w+')
```
Comment (bgruening) [inline chemicaltoolbox/rdock/select_points_SDF.py:86]: with open .... otherwise the file-handle does not get cloesed.

Comment (bgruening) [reply chemicaltoolbox/rdock/select_points_SDF.py:86]: and we looose the metadata here, left?

File: chemicaltoolbox/rdock/select_points_SDF.xml (hunk 7 above)
Comment (simonbray) [reply chemicaltoolbox/rdock/select_points_SDF.xml:55]: rdock citation. Strange... it works fine for me: https://doi.org/10.1371/journal.pcbi.1003571/

File: chemicaltoolbox/rdock/select_points_SDF.py (hunk 8 above)
Comment (simonbray) [reply chemicaltoolbox/rdock/select_points_SDF.py]: oh yes good catch

**Comment by bgruening** [comment]
please add it to the tool list

**Pull request merged**